### PR TITLE
Test cleanup and add solcover

### PR DESCRIPTION
**Pull request by kfichter**
## Description
This PR cleans up our tests further and adds `solcover` as a way to check the test coverage of our contracts. Changes are relatively minor, though require modifications to lots of different files. Sorry for the big diff! This is the last cleanup PR before we start to add new tests and contracts.

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

**Comment by kfichter** [comment]
Note that the only new functionality within this PR is in: https://github.com/ethereum-optimism/optimism-monorepo/pull/178/files#diff-e097963e3c5e4ab06efcc631731c1aab

File: packages/contracts/.solcover.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const path = require('path')
```
Comment (kfichter) [inline packages/contracts/.solcover.js:1]: General explainer for this file: `solcover` only gives us the option to skip specific contracts during the instrumentation phase (so we can't explicitly list out the contracts we *do* want). `solcover` also doesn't allow us to use globs. This file pulls out the `testfiles` argument from the coverage script inside `package.json` and automatically determines the corresponding contract. It then skips all contracts *except* the corresponding contract.

File: packages/contracts/test/test-helpers/constants.ts [hunk 2]
```diff
@@ -1,10 +1,51 @@
...
+  Opcode.ORIGIN,
+  Opcode.SELFBALANCE,
+  Opcode.SELFDESTRUCT,
+  Opcode.SLOAD,
+  Opcode.SSTORE,
+  Opcode.STATICCALL,
+  Opcode.TIMESTAMP,
+]
+
+export const GAS_LIMIT = 1_000_000_000
```
Comment (karlfloersch) [inline packages/contracts/test/test-helpers/constants.ts:44]: Someday we won't default to a billion gas 😅

File: packages/contracts/test/test-helpers/ovm-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+ * parse them, and then convert them into logs which look like they would if you were running this tx
+ * using an OVM backend.
+ *
+ * NOTE: The input logs MUST NOT be stripped of any Execution Manager events, or this function will break.
+ *
+ * @param logs An array of internal transaction logs which we will parse and then convert.
+ * @param executionManagerAddress The address of the Execution Manager contract for log parsing.
+ * @return the converted logs
+ */
+export const convertInternalLogsToOvmLogs = (
```
Comment (karlfloersch) [inline packages/contracts/test/test-helpers/ovm-helpers.ts:41]: At some point I believe this should be moved into the `rollup-fullnode` package (or deleted generally) cc @K-Ho

File: packages/core-db/test/app/ethereum/contracts/TestToken.json [hunk 4]
```diff
@@ -0,0 +1,259 @@
+{
```
Comment (karlfloersch) [inline packages/core-db/test/app/ethereum/contracts/TestToken.json:1]: Oh this should be removed like it was in this PR -- https://github.com/ethereum-optimism/optimism-monorepo/pull/173

File: yarn.lock [hunk 5]
```diff
@@ -39,18 +39,18 @@
   resolved "https://registry.yarnpkg.com/@ensdomains/resolver/-/resolver-0.2.4.tgz#c10fe28bf5efbf49bff4666d909aed0265efbc89"
```
Comment (karlfloersch) [inline yarn.lock:39]: Wow pretty large yarn.lock!

File: packages/contracts/.solcover.js (hunk 1 above)
Comment (karlfloersch) [reply packages/contracts/.solcover.js:1]: Could consider putting this explainer in as a comment in the file

File: yarn.lock (hunk 5 above)
Comment (kfichter) [reply yarn.lock:39]: Yeah unfortunately this is a side-effect of buidler for now.

**Pull request merged**